assets/src/editor/events/ticket-assignments-manager/ticket-assignments-manager-modal.js

```diff
@@ -41,463 +41,180 @@ const {
-			const newState = handler.assignTicketEntity( prevState, dateEntity, ticketEntity );
-			return { formError: '', ...newState };
-		} );
-	};
-
-	/**
-	 * removes ticket relation from date
-	 *
-	 * @function
-	 * @param {Object} dateEntity
-	 * @param {Object} ticketEntity
-	 */
-	removeTicketEntity = ( dateEntity, ticketEntity ) => {
-		if (
-			! isModelEntityOfModel( dateEntity, 'datetime' ) ||
-			! isModelEntityOfModel( ticketEntity, 'ticket' )
-		) {
-			return;
-		}
-		this.setState( ( prevState ) => {
-			const newState = handler.removeTicketEntity( prevState, dateEntity, ticketEntity );
-			return { formError: '', ...newState };
-		} );
-	};
-
-	/**
-	 * counts number of relations between tickets and dates
-	 * and builds an object with those counts indexed by object ID
-	 *
-	 * @function
-	 * @param {Array} dateEntities
-	 * @param {Array} ticketEntities
-	 * @param {Object} ticketEntitiesByDateIds
-	 * @return {number} noAssignments
-	 */
-	countTicketAssignments = ( dateEntities, ticketEntities, ticketEntitiesByDateIds ) => {
-		let dateTickets = [];
-		let ticketDateCount = 0;
-		let dateTicketCount = 0;
-		let noAssignments = 0;
-		dateEntities.forEach( ( eventDate ) => {
-			dateTicketCount = 0;
-			warning(
-				isModelEntityOfModel( eventDate, 'datetime' ),
-				'Invalid EE Date model object!'
-			);
-			dateTickets = ticketEntitiesByDateIds[ eventDate.id ] ?
-				ticketEntitiesByDateIds[ eventDate.id ] :
-				[];
-			this.assignmentCounts.dateEntities[ eventDate.id ] = dateTickets.length;
-			ticketEntities.forEach( ( ticket ) => {
-				warning(
-					isModelEntityOfModel( ticket, 'ticket' ),
-					'Invalid EE' +
-					' Ticket model object!'
+	}, [
+		assignedState,
+		addTicketEntities,
+		removeTicketEntities,
+		hasNoAssignments,
+	] );
+	const assignTicketEntity = useCallback(
+		( dateId, ticketId ) => {
+			setAssignedState( ( prevState ) => {
+				return handler.assignTicketEntity(
+					cloneDeep( prevState ),
+					dateId,
+					ticketId
 				);
-				if ( typeof this.assignmentCounts.ticketEntities[ ticket.id ] === 'undefined' ) {
-					this.assignmentCounts.ticketEntities[ ticket.id ] = 0;
-				}
-				const ticketAssignedToDate = findIndex(
-					dateTickets,
-					{ id: ticket.id }
-				) > noIndex;
-				if ( ticketAssignedToDate ) {
-					dateTicketCount++;
-					ticketDateCount++;
-					this.assignmentCounts.ticketEntities[ ticket.id ]++;
-				}
 			} );
-			if (
-				dateTicketCount === 0 &&
-				this.ticketCount > 1 &&
-				! this.state.assigned[ eventDate.id ]
-			) {
-				noAssignments++;
-			}
-		} );
-		if ( this.ticketCount === 1 && ticketDateCount > 0 ) {
-			noAssignments = 0;
-		}
-		return noAssignments;
-	};
-
+			setFormError( '' );
+		},
+		[ setFormError ]
+	);
+	const removeTicketEntity = useCallback(
+		( dateId, ticketId ) => {
+			setAssignedState( ( prevState ) => {
+				return handler.removeTicketEntity(
+					cloneDeep( prevState ),
+					dateId,
+					ticketId
+				);
+			} );
+		},
+		[]
```

Comment:
This does not need to have any dependencies because the state setter does not change.